woocommerce/woocommerce-android
# [Custom Amounts M3] Order details design changes

Closes: #10162
Closes: #9942 


### Description

This PR adds design changes on the Order detail screen as per the design specification ichqhxDjbmZZmvJaJkGhkj-fi-118%3A6643

Prominent changes are:
1. Custom Amount section in the Order detail screen
2. All the section headers are now changed as per design
3. The payment section will be hidden if there are no products or custom amount
4. Customer note, Shipping address, and Billing address section headers are shown only when there are contents

### Testing instructions

1. Navigate to the order details screen
2. **Ensure you see the custom amounts section if there are any custom amounts added for that order**
3. **Ensure all the headings are as per design**
4. **Ensure the payment sub-sections (discounts, tax, shipping ...etc) will be hidden if there are no products or custom amounts added.**
5. **Ensure customer note, Shipping address, and Billing address section headers are shown only when there are contents**

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/1331230/93453538-922a-486e-b8b5-58250a1f7bbd
